pkg/provider/fake/fake.go

```diff
@@ -124,16 +127,47 @@ func (p *Provider) PushSecret(_ context.Context, value []byte, _ corev1.SecretTy
 	return nil
 }
 
-// Empty GetAllSecrets.
-func (p *Provider) GetAllSecrets(_ context.Context, _ esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
-	// TO be implemented
-	return nil, fmt.Errorf("GetAllSecrets not implemented")
+// GetAllSecrets returns multiple secrets from the given ExternalSecretFind
+// Currently, only the Name operator is supported.
+func (p *Provider) GetAllSecrets(_ context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	if ref.Name != nil {
+		matcher, err := find.New(*ref.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		latestDataMap := make(map[string]*Data)
+		for key, data := range p.config {
```

Comment:
This should be of type `[]byte` since that's the only thing you end up using eventually anyways. Then the convert below this into a new map would not be needed anymore.

Also, take care of reassigning the data value to an in loop value if your are going to dereference it to avoid the notorious for loop variable bug.🐞

Reply:
Thanks for your feedback, @Skarlso!

> This should be of type []byte since that's the only thing you end up using eventually anyways. Then the convert below this into a new map would not be needed anymore.

We actually need to compare versions to return only the latest ones, and thus I need the first loop. Please take a look at the [L150](https://github.com/external-secrets/external-secrets/pull/2844/files#diff-f9574100ef93866473d5d538b2689fee8e22809c54295a76cb1fe4ad3f7ee624R150-R152) for more details 🙂

> Also, take care of reassigning the data value to an in loop value if your are going to dereference it to avoid the notorious for loop variable bug.🐞

Makes sense. I'll create a new variable from the key instead of reassigning it. Thanks!

Reply:
>We actually need to compare versions to return only the latest ones, and thus I need the first loop. Please take a look at the [L150]

Sure, but you aren't storing that or using it further.

Reply:
Oh sorry, I see what you mean. :D